rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java

```diff
@@ -26,215 +26,226 @@
+            sublistByWrittenKey.putIfAbsent(key, sublist);
         }
     }
 
-    private Optional<TransactionBucket> getBucketBySender(Transaction tx) {
-        return Optional.ofNullable(bucketBySender.get(tx.getSender()));
+    private Optional<TransactionSublist> getSublistBySender(Transaction tx) {
+        return Optional.ofNullable(sublistBySender.get(tx.getSender()));
     }
 
-    private Optional<TransactionBucket> getAvailableBucketWithLessUsedGas(long txGasLimit) {
+    private Optional<TransactionSublist> getAvailableSublistWithLessUsedGas(long txGasLimit) {
         long gasUsed = Long.MAX_VALUE;
-        Optional<TransactionBucket> bucketCandidate =  Optional.empty();
+        Optional<TransactionSublist> sublistCandidate =  Optional.empty();
 
-        for (TransactionBucket bucket : buckets) {
-            if (!bucket.isSequential() && bucket.hasGasAvailable(txGasLimit) && bucket.getGasUsed() < gasUsed) {
-                    bucketCandidate = Optional.of(bucket);
-                    gasUsed = bucket.getGasUsed();
+        for (TransactionSublist sublist : sublists) {
+            if (!sublist.isSequential() && sublist.hasGasAvailable(txGasLimit) && sublist.getGasUsed() < gasUsed) {
+                    sublistCandidate = Optional.of(sublist);
+                    gasUsed = sublist.getGasUsed();
             }
         }
 
-        return bucketCandidate;
+        return sublistCandidate;
     }
 
 
-    private TransactionBucket getBucketCandidates(Transaction tx, Set<ByteArrayWrapper> newReadKeys, Set<ByteArrayWrapper> newWrittenKeys) {
-        Optional<TransactionBucket> bucketCandidate = getBucketBySender(tx);
+    private TransactionSublist getSublistCandidates(Transaction tx, Set<ByteArrayWrapper> newReadKeys, Set<ByteArrayWrapper> newWrittenKeys) {
+        Optional<TransactionSublist> sublistCandidate = getSublistBySender(tx);
 
-        if (bucketCandidate.isPresent() && bucketCandidate.get().isSequential()) {
-            return getSequentialBucket();
+        if (sublistCandidate.isPresent() && sublistCandidate.get().isSequential()) {
+            return getSequentialSublist();
         }
 
         // read - written
         for (ByteArrayWrapper newReadKey : newReadKeys) {
-            if (bucketByWrittenKey.containsKey(newReadKey)) {
-                TransactionBucket bucket = bucketByWrittenKey.get(newReadKey);
-
-                if (bucketCandidate.isPresent() && !bucketCandidate.get().equals(bucket)) {
-                   return getSequentialBucket();
-                } else if (!bucketCandidate.isPresent()) {
-                    bucketCandidate = Optional.of(bucket);
-                }
+            if (sublistByWrittenKey.containsKey(newReadKey)) {
+                TransactionSublist sublist = sublistByWrittenKey.get(newReadKey);
+                sublistCandidate = Optional.of(sublistCandidate.map(sc -> returnsSequentialIfBothAreDifferent(sc, sublist)).orElse(sublist));
```

Comment:
In my opinion, the previous version of this was more clear and easier to understand even if it requires more lines of code.